topostats/processing.py

```diff
@@ -199,81 +198,140 @@ def run_grains(  # noqa: C901
                 for direction, image_arrays in grains.directions.items():
                     LOGGER.debug(f"[{filename}] : Plotting {direction} Grain Finding Images")
                     grain_out_path_direction = grain_out_path / f"{direction}"
+                    # Possibly delete this creation of the directory since we already do this earlier?
                     if plotting_config["image_set"] == "all":
-                        grain_out_path_direction.mkdir(parents=True, exist_ok=True)
```

Comment:
Not checked in detail but are we sure the `grain_out_path_direction` will exist and this line is no longer needed?